### ISSUE TITLE
Wrong type when accessing some "property" attributes from a TypeVar bounded to type in 1.1.283

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#4291]
```python
from typing import TypeVar

_T = TypeVar("_T", bound=type)

def foo(klass: _T):
    reveal_type(klass.__dict__)
    klass.__dict__.get("foo")
```

The output from this is:

```
pyright 1.1.283
/tmp/xxx.py
  /tmp/xxx.py:8:17 - information: Type of "klass.__dict__" is "property*"
  /tmp/xxx.py:9:20 - error: Cannot access member "get" for type "property*"
    Member "get" is unknown (reportGeneralTypeIssues)
  /tmp/xxx.py:7:16 - warning: TypeVar "_T" appears only once in generic function signature (reportInvalidTypeVarUse)
1 error, 1 warning, 1 information 
Completed in 0.572sec
```

This also happens when accessing some other attributes, like `__mro__`, etc.

**Comment by rsokl** [comment]
This need not involve a TypeVar, the same thing occurs for the annotation `klass: type`.

**Comment by bellini666** [comment]
> This need not involve a TypeVar, the same thing occurs for the annotation klass: type.

Oh, you are correct. I tested with `Type` and the issue did not occur, but it does indeed happens with `type`

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.284, which I just published. It will also be included in a future release of pylance.
